deepchem/utils/pdbqt_utils.py

```diff
@@ -0,0 +1,315 @@
+      "[N,O,S])&!$([CD3](=[N+])-!@[#7!D1])&!$([#7!D1]-!@[CD3]=[N+])]-!@[!$(*#"
+      "*)&!D1&!$(C(F)(F)F)&!$(C(Cl)(Cl)Cl)&!$(C(Br)(Br)Br)&!$(C([CH3])([CH3])"
+      "[CH3])]")
+  rdmolops.FastFindRings(mol)
+  rotatable_bonds = mol.GetSubstructMatches(pattern)
+  return rotatable_bonds
+
+
+def convert_mol_to_pdbqt(mol, outfile):
+  """Writes the provided ligand molecule to specified file in pdbqt format.
+
+  Creates a torsion tree and write to pdbqt file. The torsion tree
+  represents rotatable bonds in the molecule. 
+
+  Note
+  ----
+  This function requires RDKit to be installed.
+
+  Parameters
+  ----------
+  mol: rdkit Mol
+    The molecule whose value is stored in pdb format in outfile
+  outfile: str
+    Filename for a valid pdb file with the extention .pdbqt
+  """
+  import networkx as nx
+  # Walk through the original file and extract ATOM/HETATM lines and
+  # add PDBQT charge annotations.
+  pdb_map = _create_pdb_map(outfile)
+  graph = mol_to_graph(mol)
+  rotatable_bonds = get_rotatable_bonds(mol)
+
+  # Remove rotatable bonds from this molecule
+  for bond in rotatable_bonds:
+    graph.remove_edge(bond[0], bond[1])
+  # Get the connected components now that the rotatable bonds have
+  # been removed.
+  components = [x for x in nx.connected_components(graph)]
+  comp_map = _create_component_map(mol, components)
+
+  used_partitions = set()
+  lines = []
+  # The root is the largest connected component.
+  root = max(enumerate(components), key=lambda x: len(x[1]))[0]
+  # Write the root component
+  lines.append("ROOT\n")
+  for atom in components[root]:
+    lines.append(pdb_map[atom])
+  lines.append("ENDROOT\n")
+  # We've looked at the root, so take note of that
+  used_partitions.add(root)
+  for bond in rotatable_bonds:
+    valid, next_partition = _valid_bond(used_partitions, bond, root, comp_map)
+    if not valid:
+      continue
+    _dfs(used_partitions, next_partition, bond, components, rotatable_bonds,
+         lines, pdb_map, comp_map)
+  lines.append("TORSDOF %s" % len(rotatable_bonds))
+  with open(outfile, 'w') as fout:
+    for line in lines:
+      fout.write(line)
+
+
+def _create_pdb_map(outfile):
+  """Create a mapping from atom numbers to lines to write to pdbqt
+
+  This is a map from rdkit atom number to its line in the pdb
+  file. We also add the two additional columns required for
+  pdbqt (charge, symbol).
+
+  Note rdkit atoms are 0 indexed and pdb files are 1 indexed
+
+  Parameters
+  ----------
+  outfile: str
+    filename which already has a valid pdb representation of mol
+
+  Returns
+  -------
+  pdb_map: dict
+    Maps rdkit atom numbers to lines to be written to PDBQT file.
+  """
+  lines = [x.strip() for x in open(outfile).readlines()]
+  lines = filter(lambda x: x.startswith("HETATM") or x.startswith("ATOM"),
+                 lines)
+  lines = [x[:66] for x in lines]
+  pdb_map = {}
+  for line in lines:
+    my_values = line.split()
```

Comment:
PDB files are based on fixed column positions.  They are not whitespace delimited.  See http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM.

Reply:
Good catch! Fixed this to use the correct column numbers.